vm/parser.ts

```diff
@@ -6,20 +6,62 @@
 module J2ME {
   declare var util;
   import assert = J2ME.Debug.assert;
+  import concat5 = StringUtilities.concat5;
+  import concat3 = StringUtilities.concat3;
+  import pushMany = ArrayUtilities.pushMany;
+  import unique = ArrayUtilities.unique;
+  import hashBytesTo32BitsMurmur = HashUtilities.hashBytesTo32BitsMurmur;
+  export enum UTF8Chars {
+    Z = 90,
+    C = 67,
+    F = 70,
+    D = 68,
+    B = 66,
+    S = 83,
+    I = 73,
+    J = 74,
+    V = 86,
+
+    L = 76,
+    OpenBracket = 91,
+    Semicolon = 59,
+    Dot = 46,
+    Slash = 47,
+
+    OpenParenthesis = 40,
+    CloseParenthesis = 41,
+
+  }
 
   module UTF8 {
     export var Code = new Uint8Array([67, 111, 100, 101]);
     export var ConstantValue = new Uint8Array([67, 111, 110, 115, 116, 97, 110, 116, 86, 97, 108, 117, 101]);
+    export var Init = new Uint8Array([60, 105, 110, 105, 116, 62]);
+    export var OpenBracket = new Uint8Array([UTF8Chars.OpenBracket]);
+    export var OpenBracketL = new Uint8Array([UTF8Chars.OpenBracket, UTF8Chars.L]);
+    export var Semicolon = new Uint8Array([UTF8Chars.Semicolon]);
+
+    export var Z = new Uint8Array([UTF8Chars.Z]);
+    export var C = new Uint8Array([UTF8Chars.C]);
+    export var F = new Uint8Array([UTF8Chars.F]);
+    export var D = new Uint8Array([UTF8Chars.D]);
+    export var B = new Uint8Array([UTF8Chars.B]);
+    export var S = new Uint8Array([UTF8Chars.S]);
+    export var I = new Uint8Array([UTF8Chars.I]);
+    export var J = new Uint8Array([UTF8Chars.J]);
+
+    // "<init>".split("").map(function (x) { return String.charCodeAt(x); })
   }
 
-  function strcmp(a: Uint8Array, b: Uint8Array): boolean {
+  export function strcmp(a: Uint8Array, b: Uint8Array): boolean {
     if (a === b) {
       return true;
     }
     if (a.length !== b.length) {
       return false;
     }
-    for (var i = 0; i < a.length; i++) {
+    var l = a.length;
+    for (var i = 0; i < l; i++) {
```

Comment:
I don't know if it does or not. I did see some comments on bugzilla about SM being able to optimize range checks if it sees the `i < a.length` pattern. I'll change it back to that.